diff --git a/include/cef_browser.h b/include/cef_browser.h
index d82f82d..3c48dd9 100644
--- a/include/cef_browser.h
+++ b/include/cef_browser.h
@@ -1987,6 +1987,7 @@ class CefBrowserHost : public virtual CefBaseRefCounted {
   /*--cef()--*/
   virtual void ScrollByWithAnime(float delta_x, float delta_y, int32_t duration) = 0;
 
+#if defined(OHOS_GET_SCROLL_OFFSET)
   ///
   /// Get the current scroll offset of the webpage.
   ///
@@ -1998,6 +1999,7 @@ class CefBrowserHost : public virtual CefBaseRefCounted {
   ///
   /*--cef()--*/
   virtual void GetOverScrollOffset(float* offset_x, float* offset_y) = 0;
+#endif
 #endif  // BUILDFLAG(IS_OHOS)
 
   ///
diff --git a/libcef/browser/browser_host_base.cc b/libcef/browser/browser_host_base.cc
index 23506b9..6b1b3f4 100644
--- a/libcef/browser/browser_host_base.cc
+++ b/libcef/browser/browser_host_base.cc
@@ -3154,6 +3154,7 @@ void CefBrowserHostBase::ScrollPageUpDown(bool is_up,
   }
 }
 
+#ifdef OHOS_GET_SCROLL_OFFSET
 void CefBrowserHostBase::GetScrollOffset(float* offset_x,
                                          float* offset_y) {
   auto frame = GetMainFrame();
@@ -3162,6 +3163,7 @@ void CefBrowserHostBase::GetScrollOffset(float* offset_x,
         ->GetScrollOffset(offset_x, offset_y);
   }
 }
+#endif
 #endif  // #ifdef OHOS_PAGE_UP_DOWN
 
 CefRefPtr<CefBinaryValue> CefBrowserHostBase::GetWebState() {
@@ -3270,6 +3272,7 @@ void CefBrowserHostBase::ScrollByWithAnime(float delta_x, float delta_y, int32_t
   }
 }
 
+#ifdef OHOS_GET_SCROLL_OFFSET
 void CefBrowserHostBase::GetOverScrollOffset(float* offset_x,
                                              float* offset_y) {
   auto frame = GetMainFrame();
@@ -3278,6 +3281,7 @@ void CefBrowserHostBase::GetOverScrollOffset(float* offset_x,
         ->GetOverScrollOffset(offset_x, offset_y);
   }
 }
+#endif
 #endif  // defined(OHOS_INPUT_EVENTS)
 
 #ifdef OHOS_NETWORK_CONNINFO
diff --git a/libcef/browser/browser_host_base.h b/libcef/browser/browser_host_base.h
index 52a22fe..e3ab6d3 100644
--- a/libcef/browser/browser_host_base.h
+++ b/libcef/browser/browser_host_base.h
@@ -453,7 +453,9 @@ bool NeedsReload() override;
 #endif
 #ifdef OHOS_PAGE_UP_DOWN
   void ScrollPageUpDown(bool is_up, bool is_half, float view_height) override;
+#ifdef OHOS_GET_SCROLL_OFFSET
   void GetScrollOffset(float* offset_x, float* offset_y) override;
+#endif
 #endif  // OHOS_PAGE_UP_DOWN
 #if defined(OHOS_INPUT_EVENTS)
   void ScrollTo(float x, float y) override;
@@ -467,7 +469,9 @@ bool NeedsReload() override;
   void UpdateDrawRect() override;
   void ScrollToWithAnime(float x, float y, int32_t duration) override;
   void ScrollByWithAnime(float delta_x, float delta_y, int32_t duration) override;
+#if defined(OHOS_GET_SCROLL_OFFSET)
   void GetOverScrollOffset(float* offset_x, float* offset_y) override;
+#endif
 #endif  // defined(OHOS_INPUT_EVENTS)
 #ifdef OHOS_NETWORK_CONNINFO
   void SetFileAccess(bool falg) override;
diff --git a/libcef/browser/frame_host_impl.cc b/libcef/browser/frame_host_impl.cc
index dd5edd8..80013d0 100644
--- a/libcef/browser/frame_host_impl.cc
+++ b/libcef/browser/frame_host_impl.cc
@@ -1054,6 +1054,7 @@ void CefFrameHostImpl::ScrollPageUpDown(bool is_up,
                         is_up, is_half, view_height));
 }
 
+#ifdef OHOS_GET_SCROLL_OFFSET
 void CefFrameHostImpl::GetScrollOffset(float* offset_x,
                                        float* offset_y) {
   SendToRenderFrame(__FUNCTION__,
@@ -1065,6 +1066,7 @@ void CefFrameHostImpl::GetScrollOffset(float* offset_x,
                         },
                         offset_x, offset_y));
 }
+#endif
 #endif  // #ifdef OHOS_PAGE_UP_DOWN
 
 #if defined(OHOS_INPUT_EVENTS)
@@ -1165,6 +1167,7 @@ void CefFrameHostImpl::ScrollByWithAnime(float delta_x, float delta_y, int32_t d
                                       delta_x, delta_y, duration));
 }
 
+#if defined(OHOS_GET_SCROLL_OFFSET)
 void CefFrameHostImpl::GetOverScrollOffset(float* offset_x,
                                            float* offset_y) {
   SendToRenderFrame(__FUNCTION__,
@@ -1176,6 +1179,7 @@ void CefFrameHostImpl::GetOverScrollOffset(float* offset_x,
                         },
                         offset_x, offset_y));
 }
+#endif
 #endif  // defined(OHOS_INPUT_EVENTS)
 
 #endif  // BUILDFLAG(IS_OHOS)
diff --git a/libcef/browser/frame_host_impl.h b/libcef/browser/frame_host_impl.h
index 8aa4b2c..c9176bc 100644
--- a/libcef/browser/frame_host_impl.h
+++ b/libcef/browser/frame_host_impl.h
@@ -213,7 +213,9 @@ class CefFrameHostImpl : public CefFrame, public cef::mojom::BrowserFrame {
   void RemoveCache(bool include_disk_files);
 #ifdef OHOS_PAGE_UP_DOWN
   void ScrollPageUpDown(bool is_up, bool is_half, float view_height);
+#ifdef OHOS_GET_SCROLL_OFFSET
   void GetScrollOffset(float* offset_x, float* offset_y);
+#endif
 #endif  // #ifdef OHOS_PAGE_UP_DOWN
 #if defined(OHOS_INPUT_EVENTS)
   void ScrollTo(float x, float y);
@@ -226,7 +228,9 @@ class CefFrameHostImpl : public CefFrame, public cef::mojom::BrowserFrame {
   void UpdateDrawRect();
   void ScrollToWithAnime(float x, float y, int32_t duration);
   void ScrollByWithAnime(float delta_x, float delta_y, int32_t duration);
+#if defined(OHOS_GET_SCROLL_OFFSET)
   void GetOverScrollOffset(float* offset_x, float* offset_y);
+#endif
 #endif  // defined(OHOS_INPUT_EVENTS)
 
 #endif  // BUILDFLAG(IS_OHOS)
diff --git a/libcef/common/mojom/cef.mojom b/libcef/common/mojom/cef.mojom
index d982e4b..043b82d 100644
--- a/libcef/common/mojom/cef.mojom
+++ b/libcef/common/mojom/cef.mojom
@@ -142,7 +142,7 @@ interface RenderFrame {
   ScrollPageUpDown(bool is_up, bool move_half, float view_height);
   // #endif OHOS_PAGE_UP_DOWN
 
-  [EnableIf=ohos_page_up_down, Sync]
+  [EnableIf=ohos_get_scroll_offset, Sync]
   GetScrollOffset() => (float offset_x, float offset_y);
 
   // #if defined(OHOS_INPUT_EVENTS)
@@ -167,7 +167,7 @@ interface RenderFrame {
   [EnableIf=ohos_input_events]
   SetOverscrollMode(int32 mode);
 
-  [EnableIf=ohos_input_events, Sync]
+  [EnableIf=ohos_get_scroll_offset, Sync]
   GetOverScrollOffset() => (float offset_x, float offset_y);
 
   [EnableIf=is_ohos]
diff --git a/libcef/renderer/frame_impl.cc b/libcef/renderer/frame_impl.cc
index 818d6e5..a475f82 100644
--- a/libcef/renderer/frame_impl.cc
+++ b/libcef/renderer/frame_impl.cc
@@ -1290,6 +1290,7 @@ void CefFrameImpl::ScrollPageUpDown(bool is_up,
                         base::Milliseconds(computeDurationInMilliSec(0, dy)));
 }
 
+#ifdef OHOS_GET_SCROLL_OFFSET
 void CefFrameImpl::GetScrollOffset(
     cef::mojom::RenderFrame::GetScrollOffsetCallback callback) {
   auto render_frame = content::RenderFrame::FromWebFrame(frame_);
@@ -1302,6 +1303,7 @@ void CefFrameImpl::GetScrollOffset(
   auto scroll_offset = webView->GetScrollOffset();
   std::move(callback).Run(scroll_offset.x(), scroll_offset.y());
 }
+#endif
 #endif  // #ifdef OHOS_PAGE_UP_DOWN
 
 
@@ -1409,6 +1411,7 @@ void CefFrameImpl::ScrollByWithAnime(float delta_x, float delta_y, int32_t durat
                         base::Milliseconds(duration));
 }
 
+#if defined(OHOS_GET_SCROLL_OFFSET)
 void CefFrameImpl::GetOverScrollOffset(
     cef::mojom::RenderFrame::GetOverScrollOffsetCallback callback) {
   auto render_frame = content::RenderFrame::FromWebFrame(frame_);
@@ -1417,6 +1420,7 @@ void CefFrameImpl::GetOverScrollOffset(
   auto overScroll_offset = render_frame->GetOverScrollOffset();
   std::move(callback).Run(overScroll_offset.x(), overScroll_offset.y());
 }
+#endif
 #endif  // defined(OHOS_INPUT_EVENTS)
 
 bool CefFrameImpl::ShouldOverrideUrlLoading(const CefString& url,
diff --git a/libcef/renderer/frame_impl.h b/libcef/renderer/frame_impl.h
index eb739e8..435b3aa 100644
--- a/libcef/renderer/frame_impl.h
+++ b/libcef/renderer/frame_impl.h
@@ -212,8 +212,10 @@ class CefFrameImpl
   void RemoveCache() override;
 #ifdef OHOS_PAGE_UP_DOWN
   void ScrollPageUpDown(bool is_up, bool is_half, float view_height) override;
+#ifdef OHOS_GET_SCROLL_OFFSET
   void GetScrollOffset(
       cef::mojom::RenderFrame::GetScrollOffsetCallback callback) override;
+#endif
 #endif  // #ifdef OHOS_PAGE_UP_DOWN
 
 #if defined(OHOS_INPUT_EVENTS)
@@ -234,8 +236,10 @@ class CefFrameImpl
   void UpdateDrawRect() override;
   void ScrollToWithAnime(float x, float y, int32_t duration) override;
   void ScrollByWithAnime(float delta_x, float delta_y, int32_t duration) override;
+#if defined(OHOS_GET_SCROLL_OFFSET)
   void GetOverScrollOffset(
       cef::mojom::RenderFrame::GetOverScrollOffsetCallback callback) override;
+#endif
 #endif  // defined(OHOS_INPUT_EVENTS)
 
   GURL GetAbsoluteUrl(const blink::WebNode& node,
diff --git a/libcef_dll/ctocpp/browser_host_ctocpp.cc b/libcef_dll/ctocpp/browser_host_ctocpp.cc
index 887f127..5167c68 100644
--- a/libcef_dll/ctocpp/browser_host_ctocpp.cc
+++ b/libcef_dll/ctocpp/browser_host_ctocpp.cc
@@ -3442,6 +3442,7 @@ void CefBrowserHostCToCpp::AutoFillWithIMFEvent(bool is_username,
                                    is_new_password, content.c_str());
 }
 
+#ifdef OHOS_GET_SCROLL_OFFSET
 NO_SANITIZE("cfi-icall")
 void CefBrowserHostCToCpp::GetScrollOffset(float* offset_x, float* offset_y) {
   shutdown_checker::AssertNotShutdown();
@@ -3472,6 +3473,7 @@ void CefBrowserHostCToCpp::GetOverScrollOffset(float* offset_x,
   // Execute
   _struct->get_overscroll_offset(_struct, offset_x, offset_y);
 }
+#endif
 // CONSTRUCTOR - Do not edit by hand.
 
 CefBrowserHostCToCpp::CefBrowserHostCToCpp() {}
diff --git a/libcef_dll/ctocpp/browser_host_ctocpp.h b/libcef_dll/ctocpp/browser_host_ctocpp.h
index e27425c..9833767 100644
--- a/libcef_dll/ctocpp/browser_host_ctocpp.h
+++ b/libcef_dll/ctocpp/browser_host_ctocpp.h
@@ -306,8 +306,10 @@ class CefBrowserHostCToCpp : public CefCToCppRefCounted<CefBrowserHostCToCpp,
                             bool is_other_account,
                             bool is_new_password,
                             const std::string& content) override;
+#ifdef OHOS_GET_SCROLL_OFFSET
   void GetScrollOffset(float* offset_x, float* offset_y) override;
   void GetOverScrollOffset(float* offset_x, float* offset_y) override;
+#endif
 };
 
 #endif  // CEF_LIBCEF_DLL_CTOCPP_BROWSER_HOST_CTOCPP_H_
