From bf8c0f35e5386aefaa810745385dd5b0226afa77 Mon Sep 17 00:00:00 2001
From: fooljohnny <fooljohnny@gmail.com>
Date: Wed, 25 Sep 2024 17:58:03 +0800
Subject: [PATCH] add ohos's custom url_utils

---
 BUILD.gn | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/BUILD.gn b/BUILD.gn
index f36b267..1962b1c 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1195,6 +1195,13 @@ source_set("libcef_static") {
     ]
   }
 
+  if (is_ohos) {
+    sources += [
+      "ohos_cef_ext/libcef/common/net/url_util_ex.h",
+      "ohos_cef_ext/libcef/common/net/url_util_ex.cc",
+    ]
+  }
+
   if (ozone_platform_x11) {
     deps += [ "//ui/events/devices/x11" ]
   }
-- 
2.25.1

